plutus-core-spec/main.tex

```diff
@@ -797,35 +797,11 @@
 
 \section{Plutus Core}
 
-Plutus Core is the target for compilation of Plutus TX, the smart
-contract language to validate transactions on Cardano. Plutus Core is
-designed to be simple and easy to reason about using proof assistants
-and automated theorem provers.
-
-Plutus Core has a strong formal basis. For those familiar with
-programming language theory, it can be described in one line: higher
-kinded System F with isorecursive types, and suitable
-primitive types and values. There are no constructs for data types,
-which are represented using Scott encodings---we do not use Church
-encodings since they may require linear rather than constant time to
-access the components of a data structure. Primitive types, such as
-Integer and Bytestring, are indexed by their size, which allows the
-cost (in gas) of operations such as addition to be determined at
-compile time. In contrast, IELE uses unbounded integers, which never
-overflow, but require that the gas used is calculated at run time. We
-use \Fomega{} (as opposed to F) because it supports types indexed by
-size and parameterised types (such as ``List of A'').
-
-Plutus Core also notable lacks obvious built-in support for recursion,
-either general or otherwise. The reason for this is simply that in
-the presence of type left fixed points, it's possible to give a valid
-type to recursion combinators. This is analogous to the fact that in
-Haskell, you can define the Y combinator but you need to use a newtype
-declaration to do. Precisely which combinators one defines is up to
-the user, but as Plutus itself is eagerly reduced, some of them will
-lead to looping behavior, including Y. Libraries can of course
-be defined with various combinators provided as a convenient way to
-gain access to recursion.
+Plutus Core is typed, eagerly evaluated $\lambda$-calculus designed for use as a transaction validation language in blockchain systems. More precisely, it is a higher-kinded variant of system F with iso-recursive types. Plutus Core is meant to be a compilation target, and this is reflected in the design of the language: while writing large Plutus Core programs by hand is difficult, the language is relatively straightforward to formalize with a proof assistant.
+
+Plutus Core includes support for \emph{sized types} and \emph{size kinds} to facilitate reasoning about running time and memory usage of programs. This is important for blockchain applications, in which programs consume \emph{gas} during execution according to their computational resource consumption.
```

Comment:
When I read "reasoning" I think about compile time rather than runtime. Approximating for how long a program will run to completion is still undecidable, right?

Reply:
additionally it might be good to mention that it's not sized types in general, but just sized data of a few sorts. otherwise people might think we mean something like agda's sized types

Reply:
@effectfully that's still undecidable yes. While the sort of reasoning we want to be able to do about programs does take place at compile time, it is the program's runtime behavior that we want to reason about. Specifically, how long it takes to run (the "running time"), and how much space it's going to use.

Reply:
@psygnisfive good point. I'll change it to "In Plutus Core, Integers and Bytestrings are \emph{size indexed}" or something.

Reply:
> it is the program's runtime behavior that we want to reason about. Specifically, how long it takes to run (the "running time"), and how much space it's going to use.

Is it possible to figure this out looking at sizes at compile time? I think we can only say "the program will fail if it consumes more than what was specified", but "reasoning" looks like is a too strong word in this case for me. When I see "reasoning", I think about static guarantees that do not result in a runtime failure.

Reply:
We can get good estimates for the time/space required for the built in functions, which is where a lot of the computational resources are going to be consumed. I suspect the best we can hope for otherwise are loose upper/lower bounds on running time and memory usage, but that's alright. I don't see how anything we're talking about could possibly result in a runtime failure.